core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -373,4 +388,62 @@ private[spark] object SparkConf {
    * Return whether the given config is a Spark port config.
    */
   def isSparkPortConf(name: String): Boolean = name.startsWith("spark.") && name.endsWith(".port")
+
+  /**
+   * Returns the configuration key to use for the given user-provided key, translating
+   * deprecated keys.
+   *
+   * @param userKey Configuration key from the user / caller.
+   * @param warn Whether to print a warning if the key is depreated. Warnings will be printed
+   *             only once for each key.
+   */
+  def getConfKey(userKey: String, warn: Boolean = false): String = {
```

Comment:
It's a little weird for this to be `getConfKey(key)`. From the name it looks like an identity function. I think a better name is `translateConfKey`